### PR TITLE
Added support for pull policy and docs

### PR DESCRIPTION
**Pull request by iuliana**
As the title says.

### CODE REVIEW

**Review by algairim — APPROVED** [review]

File: software/base/src/main/java/org/apache/brooklyn/tasks/kubectl/JobBuilder.java [hunk 1]
```diff
@@ -121,6 +130,11 @@ public String build(){
             containerSpec.setWorkingDir(workingDir);
         }
         containerSpec.setImage(imageName);
+        if(Strings.isNonBlank(imagePullPolicy)) {
+            if (PULL_POLICY_ALLOWED_VALUES.contains(imagePullPolicy)) {
+                containerSpec.setImagePullPolicy(imagePullPolicy);
+            } // else stick with default, do not fail
```
Comment (geomacy) [inline software/base/src/main/java/org/apache/brooklyn/tasks/kubectl/JobBuilder.java:136]: surely it would be better to fail?

Comment (algairim) [reply software/base/src/main/java/org/apache/brooklyn/tasks/kubectl/JobBuilder.java:136]: @geomacy, the default policy that `kubectl` sets is 'Always'. And now, we simply allow to configure it.

Comment (algairim) [reply software/base/src/main/java/org/apache/brooklyn/tasks/kubectl/JobBuilder.java:136]: @iuliana, to remove that code comment `do not fail`?

Comment (ahgittin) [reply software/base/src/main/java/org/apache/brooklyn/tasks/kubectl/JobBuilder.java:136]: agree - i'd say either:

* always set the user-supplied value, let kubectl fail if it is invalid (good if they might add new valid args)
* or if we are going to try to duplicate enforcement, use an enum and specify that as the type of the config key (allows type introspection to determine valid values)

silently ignoring is going to confuse a user (esp if kubectl is case insensitive and we just drop things that don't match our arbitrary case!)

File: software/base/README.adoc [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+////
+
+= Apache Brooklyn Software Components
```
Comment (ahgittin) [inline software/base/README.adoc:20]: this belongs in `brooklyn-docs` project not here, no?

**Comment by ahgittin** [comment]
good additions but a few comments.

i also wonder if there is mileage in an "extra_kubectl_args" list config key for other things we don't support or which get added in future versions?  (that would have allowed pull policy to be specified without code changes here.)

**Comment by iuliana** [comment]
> 

@ahgittin this should like a good idea, but the Job template has multiple levels. There are properties that you can add for `containers`, `spec`, `template` , etc.  `extra_kubectl_args` then would have to be a multi-level map, to cover these cases and this might complicate the `JobBuilder` implementation.  I guess it could be done, if really necessary. Maybe keep this for later after we have the core functionality working.

File: software/base/src/main/java/org/apache/brooklyn/tasks/kubectl/JobBuilder.java (hunk 1 above)
Comment (iuliana) [reply software/base/src/main/java/org/apache/brooklyn/tasks/kubectl/JobBuilder.java:136]: @ahgittin `enum` it is then.

Comment (geomacy) [reply software/base/src/main/java/org/apache/brooklyn/tasks/kubectl/JobBuilder.java]: > [ahgittin](https://github.com/ahgittin) [2 days ago](https://github.com/apache/brooklyn-server/pull/1329#discussion_r910735955)  silently ignoring is going to confuse a user (esp if kubectl is case insensitive and we just drop things that don't match our arbitrary case!)

That was my thought.

> [iuliana](https://github.com/iuliana) [2 days ago](https://github.com/apache/brooklyn-server/pull/1329#discussion_r911071862)
@geomacy Added enum for imagePullPolicy that validates the value, before the task is even created.

looks great thanks @iuliana

**Review by geomacy — APPROVED** [review]
LGTM

**Pull request merged**